src/DGtal/topology/CubicalComplex.h

```diff
@@ -0,0 +1,1305 @@
+  public:
+    typedef CubicalComplex< TKSpace, TCellContainer > Self;
+    
+    BOOST_CONCEPT_ASSERT(( concepts::CCellularGridSpaceND< TKSpace > ));
+    BOOST_STATIC_ASSERT( IsPairAssociativeContainer< TCellContainer >::value );
+
+    // JOL: Not used, because unordered_set and unordered_map do not pass
+    // these concept checks.
+    // BOOST_CONCEPT_ASSERT(( boost::AssociativeContainer< TCellContainer > ));
+    // BOOST_CONCEPT_ASSERT(( boost::PairAssociativeContainer< TCellContainer > ));
+
+    friend Self& DGtal::functions::operator|=<>( Self&, const Self& );
+    friend Self& DGtal::functions::operator&=<>( Self&, const Self& );
+    friend Self& DGtal::functions::operator^=<>( Self&, const Self& );
+    friend Self& DGtal::functions::operator-=<>( Self&, const Self& );
+    friend Self  DGtal::functions::operator| <>( const Self&, const Self& );
+    friend Self  DGtal::functions::operator& <>( const Self&, const Self& );
+    friend Self  DGtal::functions::operator^ <>( const Self&, const Self& );
+    friend Self  DGtal::functions::operator- <>( const Self&, const Self& );
+    friend Self  DGtal::functions::operator~ <>( const Self& );
+    friend Self  DGtal::functions::operator* <>( const Self& );
+    friend bool  DGtal::functions::operator==<>( const Self&, const Self& );
+    friend bool  DGtal::functions::operator!=<>( const Self&, const Self& );
+    friend bool  DGtal::functions::operator<=<>( const Self&, const Self& );
+    friend bool  DGtal::functions::operator>=<>( const Self&, const Self& );
+
+    typedef TKSpace KSpace;
+    typedef TCellContainer CellContainer;
+    typedef typename CellContainer::mapped_type Data;
+
+    BOOST_STATIC_ASSERT (( boost::is_base_of< CubicalCellData, Data >::value ));
+    BOOST_STATIC_ASSERT (( boost::is_same< typename TKSpace::Cell, typename CellContainer::key_type >::value ));
+
+
+    /// The dimension of the embedding space.
+    static const Dimension dimension = KSpace::dimension;
+    typedef typename KSpace::Integer     Integer;
+    typedef typename KSpace::Cell        Cell;
+    typedef typename KSpace::Cells       Cells;
+    typedef typename KSpace::Space       Space;
+    typedef typename KSpace::Size        Size;
+    typedef typename KSpace::Point       Point;
+    typedef typename KSpace::DirIterator DirIterator;
+    typedef CellContainer                CellMap;
+    typedef typename CellMap::const_iterator CellMapConstIterator;
+    typedef typename CellMap::iterator   CellMapIterator;
+
+
+    /// Possible cell types within a complex. 
+    enum CellType { 
+      Maximal /**< The cell has no proper coface */, 
+      Free    /**< The cell has 1 proper coface */,
+      Any     /**< The cell has strictly more than 2 proper cofaces.*/
+    };
+
+    /// Flag Used to indicate in a cell data that this cell has been (virtually) removed.
+    static const uint32_t REMOVED     = 0x10000000;
```

Comment:
shouldn't you use ```BOOST_STATIC_CONSTANT```? (http://www.boost.org/doc/libs/1_31_0/more/int_const_guidelines.htm)